src/prototype.rs

```diff
@@ -675,6 +677,56 @@ fn dump_tree(nodes: &Vec<Node<Item>>, mut ix: usize, level: usize) {
     }
 }
 
+/// Determines whether the delimiter run starting at given index is
+/// left-flanking, as defined by the commonmark spec (and isn't introword
+/// for _ delims)
+fn delim_run_can_open(bytes: &[u8], run_len: usize, ix: usize) -> bool {
+    if ix + run_len >= bytes.len() {
+        return false;
+    }
+    let next_byte = bytes[ix + run_len];
+
+    if is_ascii_whitespace(next_byte) {
+        return false;
+    }
+    if ix == 0 {
+        return true;
+    }
+    let delim = bytes[ix];
+    if delim == b'*' && !is_ascii_punctuation(next_byte) {
+        return true;
+    }
+
+    // FIXME: we really should get the prev char!
```

Comment:
Yup. I'm not sure off the top of my head whether there's code for this in pulldown-cmark. I know I've written it but it might be in xi :)

That said, if this function is changed to take `&str` rather than `&[u8]`, then you can use the stdlib and do it with `s[..ix].chars().rev().next()`. Similarly for next_char (don't use `rev`).

Expanding to Unicode whitespace here can be filed as an issue if it's not properly tested in the spec, and not block this PR.

Reply:
It is tested in the spec and we can get 1 or 2 more passes by fixing it :-) I'll try to do it in this pr.